web/pdf_page_view.js

```diff
@@ -667,12 +667,12 @@ class PDFPageView {
             this.maxCanvasPixels;
         }
       }
-      const postponeDrawing =
-        !onlyCssZoom && drawingDelay >= 0 && drawingDelay < 1000;
+      const postponeDrawing = drawingDelay >= 0 && drawingDelay < 1000;
```

Comment:
This is not directly related to this patch, but this code somewhat confuses me. AFAICT `drawingDelay` originates from the `defaultZoomDelay` viewer option, see: https://github.com/mozilla/pdf.js/blob/44b7cc517de7c2cb428f67a5db232a6c56675999/web/app_options.js#L119-L123
However, that doesn't specify any maximum value yet here we seem to only respect the value if it's below 1000. Moreover, if the value is 0 we also postpone drawing because of `>=` even though I'd think in that case we don't have to postpone at all.

Do you happen to know what the rationale for the current code is? As said it doesn't have to be changed in this patch, but I figured I should ask because it felt a bit odd to me.

Reply:
This is implemented to delay the actual re-rendering during quick zooming, e.g. with the mouse-wheel or touch gestures, since that improves performance.
The reason that we cap the value is to prevent the user from essentially breaking things by setting the pref to a ridiculously large value.
As far as the zero-case goes, since setTimeout is being used even `0` is actually a delay (albeit perhaps not that useful); note https://github.com/mozilla/pdf.js/blob/7e5e06b88639fbe4c2034809a56922a3f56fca30/web/pdf_viewer.js#L1248-L1253

Hopefully this helps clear up some of confusion, if not please ask again :-)

Reply:
Yes, that does clear it up; thanks! (I was wondering if we should document the lower and upper limits at e.g. the preference definition, but I'm not sure if it's really needed because zero or very large values should hopefully be edge cases anyway.)